### PR TITLE
Add Prettier to Theme

### LINKED ISSUE #70: Add Prettier
Let's add Prettier to this repo to be more closely aligned with theme, pulumi-service, etc and remove the need for considering code-style while making changes.

### PR DESCRIPTION
**Pull request by kimberleyamackenzie**
This PR:

1. Adds Prettier to theme, including some new handy commands (`yarn lint` and `yarn-lint-fix`) + a makefile update (`make lint`).
2. Runs Prettier on the repo.

fixes #70 

### Verification 
There should only be formatting + non-user facing changes here.  I have opened [a PR in hugo](https://github.com/pulumi/pulumi-hugo/pull/1176) that points to this PR branch for theme so we can manually verify everything is looking as it should using the preview build.  http://pulumi-hugo-origin-pr-1176-34b83c5d.s3-website.us-west-2.amazonaws.com/

### Open questions

1. Right now, we don't run any checks on PR build - do we want to run a lint check now that we have the linting functionality?  I don't want to add friction, and this repo doesn't see a ton of traffic, so I could go either way at the moment - 100% up to y'all.
2. Are there any files/folders/file types that we want Prettier to ignore, in addition to what I have already added in `.prettierignore`?

### CODE REVIEW

**Review by susanev — APPROVED** [review]
would it make sense to mention prettier and the commands in the readme of this repo?

File: .prettierrc.json [hunk 1]
```diff
@@ -0,0 +1 @@
+{}
```
Comment (cnunciato) [inline .prettierrc.json:1]: I think we want there to be some definitions here (and then reapply) -- here's what I've put in place on the Facet branch that [swaps FAST for Stencil](https://github.com/pulumi/facet/pull/16) (which I'd love to be able to merge once https://github.com/pulumi/theme/pull/45 is done), which has a few things we've already standardized on (like double-quotes, 4-space tabs and the like):

https://github.com/pulumi/facet/blob/cnunciato/swap-fast-for-stencil/.prettierrc.json

File: Makefile [hunk 2]
```diff
@@ -13,3 +13,7 @@ serve:
 .PHONY: build
 build:
 	yarn build
+
+.PHONY: lint
+build:
+	yarn lint
```
Comment (cnunciato) [inline Makefile:19]:
```suggestion
	yarn lint
	
```

**Review by cnunciato — CHANGES_REQUESTED** [review]
We should get some explicit Prettier config in there that reflects our existing styles, then reapply. Thanks for doing this!

**Comment by kimberleyamackenzie** [comment]
Cool, I updated the readme + also added a prettierrc using the same settings as the file you linked above @cnunciato and re-ran Prettier.

File: README.md [hunk 3]
```diff
@@ -65,3 +65,8 @@ replace github.com/pulumi/theme => ../theme
 **Tip:** If you run `make serve` (in this repo) in a terminal tab and run your Hugo dev server in another, the changes you make to CSS and JavaScript source files will be recompiled and reloaded in the browser automatically.
 
 Be sure to remove the `replace` line before you commit.
+
+### Linting
+This repo uses Prettier for code formatting, to keep styling + formatting issues aligned without the need to be reviewed in PRs.  We currently do not automtically run the formatted as a build check, but may opt to do so in the future as needed.  To see what files have formatting diffs, you can run `yarn lint` from the project root, which will list all the files with diffs.  To fix this diffs in bulk, you can run `yarn lint-fix` also from the root.
```
Comment (cnunciato) [inline README.md:70]:
```suggestion
This repo uses Prettier for code formatting, to keep styling + formatting issues aligned without the need to be reviewed in PRs.  We currently do not automatically run the formatter as a build check, but may opt to do so in the future as needed.  To see what files have formatting diffs, you can run `yarn lint` from the project root, which will list all the files with diffs.  To fix these diffs in bulk, you can run `yarn lint-fix` also from the root.
```

File: README.md [hunk 4]
```diff
@@ -65,3 +65,8 @@ replace github.com/pulumi/theme => ../theme
...
+
+Prettier is easiest to use when you run it automatically on save in your editor - see the docs here: https://prettier.io/docs/en/editors.html.  If you set your editor to format on save, you should be staying aligned with formatting without needing to run any of the above commands.
```
Comment (cnunciato) [inline README.md:72]: 🙌

File: Makefile (hunk 2 above)
Comment (cnunciato) [inline Makefile:19]: I like these Makefile wrappers in general, but the README seems to suggest calling the `yarn` commands directly. Could we go one way or the other? And if we keep the Makefile, make this little fix and adjust the README to use `make lint` (and probably also `make lint-fix`)? 

```suggestion
.PHONY: lint
line:
	yarn lint
```

**Review by cnunciato — APPROVED** [review]
Yay! Thanks. Just a couple lil' things to consider.

File: Makefile (hunk 2 above)
Comment (kimberleyamackenzie) [reply Makefile]: Cool, removed this from the makefile

**Pull request merged**